### ISSUE TITLE
wwan (on usb) problem on 251rc2

### ISSUE THREAD

**Issue by galibozek** [systemd/systemd#23288]
**systemd version the issue has been seen with**
 > 251rc2-1

**Used distribution**
 > Arch Linux

**Linux kernel version used** (`uname -a`)

 > `Linux arch 5.17.5-arch1-1 #1 SMP PREEMPT Wed, 27 Apr 2022 20:56:11 +0000 x86_64 GNU/Linux`

**CPU architecture issue was seen on**
 > x86_64

**Unexpected behaviour you saw**
 > wwan fibocom l830 stop working

**Steps to reproduce the problem**
 > upgrade systemd from 250.5-1 -> 251rc2-1

**Additional program output to the terminal or log subsystem illustrating the issue**

```text
systemd[1]: Starting Modem Manager...
ModemManager[564]: <info>  ModemManager (version 1.18.8-1) starting in system bus...
ModemManager[564]: [qrtr] socket lookup from 1:0
ModemManager[564]: [qrtr] initial lookup finished
systemd[1]: Started Modem Manager.
ModemManager[564]: opening device...
ModemManager[564]: cannot connect to proxy: Could not connect: Connection refused
ModemManager[564]: spawning new mbim-proxy (try 1)...
ModemManager[564]: opening device...
ModemManager[564]: cannot connect to proxy: Could not connect: Connection refused
ModemManager[564]: spawning new mbim-proxy (try 1)...
ModemManager[564]: [/dev/wwan0mbim0] Couldn't find descriptors file, possibly not using cdc_mbim
ModemManager[564]: [/dev/wwan0mbim0] Fallback to default max control message size: 4096
ModemManager[564]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
ModemManager[564]: [/dev/cdc-wdm0] unexpected port hangup!
ModemManager[564]: [/dev/cdc-wdm0] channel destroyed
ModemManager[564]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1f.6': not supported by any plugin
ModemManager[564]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1c.4/0000:09:00.0': not supported by any plugin
ModemManager[564]: <info>  [wwan0mbim0/mbim] MBIM device is not QMI capable
ModemManager[564]: [/dev/wwan0mbim0] channel destroyed
ModemManager[564]: proxy configuration failed: closed
ModemManager[564]: <info>  [device /sys/devices/pci0000:00/0000:00:14.0/usb3/3-2] creating modem with plugin 'fibocom' and '4' ports
ModemManager[564]: <warn>  [plugin/fibocom] could not grab port wwan0mbim0: unsupported subsystem: 'wwan'
ModemManager[564]: <warn>  [plugin/fibocom] could not grab port cdc-wdm0: Cannot add port 'usbmisc/cdc-wdm0', unhandled port type
ModemManager[564]: <info>  [base-manager] modem for device '/sys/devices/pci0000:00/0000:00:14.0/usb3/3-2' successfully created
ModemManager[564]: <warn>  [modem0] couldn't start initialization: Cannot initialize: MBIM port went missing
ModemManager[564]: <warn>  [modem0] couldn't initialize: 'Modem is unusable, cannot fully initialize'
```
on 250.5 see:

```text
systemd[1]: Started Modem Manager.
ModemManager[564]: opening device...
ModemManager[564]: cannot connect to proxy: Could not connect: Connection refused
ModemManager[564]: spawning new mbim-proxy (try 1)...
ModemManager[564]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
ModemManager[564]: <info>  [cdc-wdm0/mbim] MBIM device is not QMI capable
ModemManager[564]: [/dev/cdc-wdm0] channel destroyed
ModemManager[564]: <info>  [device /sys/devices/pci0000:00/0000:00:14.0/usb3/3-2] creating modem with plugin 'fibocom' and '3' ports
ModemManager[564]: <info>  [base-manager] modem for device '/sys/devices/pci0000:00/0000:00:14.0/usb3/3-2' successfully created
ModemManager[564]: opening device...
ModemManager[564]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
ModemManager[564]: <info>  [modem0/cdc-wdm0/mbim] MBIM device is not QMI capable
ModemManager[564]: <warn>  [modem0] couldn't load carrier config: QMI PDC not supported
ModemManager[564]: <warn>  [modem0] (shared-xmm) unexpected band value: 0-2
ModemManager[564]: <warn>  [modem0] couldn't query SIM slots: OperationNotAllowed
```

**Comment by yuwata** [comment]
Could you also provide debugging logs of networkd? It can be generated with the following drop-in configs:
```
# /etc/systemd/system/systemd-networkd.service.d/override.conf
[Service]
Environment=SYSTEMD_LOG_LEVEL=debug
```
Then, please provide the result of `journalctl --no-hostname -u systemd-networkd.service -u ModemManager.service`.

**Comment by galibozek** [comment]
I'm not using systemd-networkd - disabled, inactive

**Comment by yuwata** [comment]
Is the issue really solved when you downgrade systemd to v250?
If so, are there any relevant logs in journal, e.g. from systemd-udevd.service?

**Comment by galibozek** [comment]
yes, after downgrade to 250 all works

`journalctl --no-hostname -u systemd-udevd.service -u ModemManager.service -b` on 251rc2

```maj 06 18:27:14 systemd-udevd[175]: Using default interface naming scheme 'v251'.
maj 06 18:27:20 systemd[1]: Stopping Rule-based Manager for Device Events and Files...
maj 06 18:27:20 systemd[1]: systemd-udevd.service: Deactivated successfully.
maj 06 18:27:20 systemd[1]: Stopped Rule-based Manager for Device Events and Files.
maj 06 18:27:21 systemd[1]: Starting Rule-based Manager for Device Events and Files...
maj 06 18:27:21 systemd-udevd[496]: Using default interface naming scheme 'v251'.
maj 06 18:27:21 systemd[1]: Started Rule-based Manager for Device Events and Files.
maj 06 18:27:21 systemd[1]: Starting Modem Manager...
maj 06 18:27:21 ModemManager[555]: <info>  ModemManager (version 1.18.8-1) starting in system bus...
maj 06 18:27:21 ModemManager[555]: [qrtr] socket lookup from 1:0
maj 06 18:27:21 ModemManager[555]: [qrtr] initial lookup finished
maj 06 18:27:21 systemd[1]: Started Modem Manager.
maj 06 18:27:21 mtp-probe[635]: checking bus 3, device 2: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-2"
maj 06 18:27:21 mtp-probe[634]: checking bus 3, device 3: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-3"
maj 06 18:27:21 mtp-probe[635]: bus: 3, device: 2 was not an MTP device
maj 06 18:27:22 mtp-probe[634]: bus: 3, device: 3 was not an MTP device
maj 06 18:27:22 mtp-probe[662]: checking bus 3, device 4: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-4"
maj 06 18:27:22 mtp-probe[662]: bus: 3, device: 4 was not an MTP device
maj 06 18:27:22 mtp-probe[663]: checking bus 3, device 5: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-9"
maj 06 18:27:22 mtp-probe[663]: bus: 3, device: 5 was not an MTP device
maj 06 18:27:23 systemd-udevd[519]: cfg80211: Process '/usr/bin/set-wireless-regdom' failed with exit code 1.
maj 06 18:27:23 systemd-udevd[523]: could not read from '/sys/module/pcc_cpufreq/initstate': No such device
maj 06 18:27:24 ModemManager[555]: opening device...
maj 06 18:27:24 ModemManager[555]: cannot connect to proxy: Could not connect: Connection refused
maj 06 18:27:24 ModemManager[555]: spawning new mbim-proxy (try 1)...
maj 06 18:27:24 ModemManager[555]: opening device...
maj 06 18:27:24 ModemManager[555]: cannot connect to proxy: Could not connect: Connection refused
maj 06 18:27:24 ModemManager[555]: spawning new mbim-proxy (try 1)...
maj 06 18:27:24 ModemManager[555]: [/dev/wwan0mbim0] Couldn't find descriptors file, possibly not using cdc_mbim
maj 06 18:27:24 ModemManager[555]: [/dev/wwan0mbim0] Fallback to default max control message size: 4096
maj 06 18:27:24 ModemManager[555]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
maj 06 18:27:24 ModemManager[555]: [/dev/cdc-wdm0] unexpected port hangup!
maj 06 18:27:24 ModemManager[555]: [/dev/cdc-wdm0] channel destroyed
maj 06 18:27:25 ModemManager[555]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1f.6': not supported by any plugin
maj 06 18:27:25 ModemManager[555]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1c.4/0000:09:00.0': not supported by any plugin
maj 06 18:27:44 ModemManager[555]: <info>  [wwan0mbim0/mbim] MBIM device is not QMI capable
maj 06 18:27:44 ModemManager[555]: [/dev/wwan0mbim0] channel destroyed
```

`journalctl --no-hostname -u systemd-udevd.service -u ModemManager.service -b` on 250.5

```maj 06 18:23:36 systemd[1]: Stopping Rule-based Manager for Device Events and Files...
maj 06 18:23:36 systemd[1]: systemd-udevd.service: Deactivated successfully.
maj 06 18:23:36 systemd[1]: Stopped Rule-based Manager for Device Events and Files.
maj 06 18:23:37 systemd[1]: Started Rule-based Manager for Device Events and Files.
maj 06 18:23:37 systemd[1]: Starting Modem Manager...
maj 06 18:23:37 ModemManager[672]: <info>  ModemManager (version 1.18.8-1) starting in system bus...
maj 06 18:23:37 ModemManager[672]: [qrtr] socket lookup from 1:0
maj 06 18:23:37 ModemManager[672]: [qrtr] initial lookup finished
maj 06 18:23:37 systemd[1]: Started Modem Manager.
maj 06 18:23:37 mtp-probe[729]: checking bus 3, device 2: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-2"
maj 06 18:23:37 mtp-probe[729]: bus: 3, device: 2 was not an MTP device
maj 06 18:23:37 mtp-probe[736]: checking bus 3, device 3: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-3"
maj 06 18:23:37 mtp-probe[762]: checking bus 3, device 4: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-4"
maj 06 18:23:37 mtp-probe[762]: bus: 3, device: 4 was not an MTP device
maj 06 18:23:37 mtp-probe[736]: bus: 3, device: 3 was not an MTP device
maj 06 18:23:37 mtp-probe[790]: checking bus 3, device 5: "/sys/devices/pci0000:00/0000:00:14.0/usb3/3-9"
maj 06 18:23:37 mtp-probe[790]: bus: 3, device: 5 was not an MTP device
maj 06 18:23:38 systemd-udevd[623]: cfg80211: Process '/usr/bin/set-wireless-regdom' failed with exit code 1.
maj 06 18:23:38 systemd-udevd[608]: Using default interface naming scheme 'v250'.
maj 06 18:23:38 systemd-udevd[613]: Using default interface naming scheme 'v250'.
maj 06 18:23:38 systemd-udevd[626]: Using default interface naming scheme 'v250'.
maj 06 18:23:39 systemd-udevd[629]: Using default interface naming scheme 'v250'.
maj 06 18:23:40 ModemManager[672]: opening device...
maj 06 18:23:40 ModemManager[672]: cannot connect to proxy: Could not connect: Connection refused
maj 06 18:23:40 ModemManager[672]: spawning new mbim-proxy (try 1)...
maj 06 18:23:40 ModemManager[672]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
maj 06 18:23:40 ModemManager[672]: <info>  [cdc-wdm0/mbim] MBIM device is not QMI capable
maj 06 18:23:40 ModemManager[672]: [/dev/cdc-wdm0] channel destroyed
maj 06 18:23:41 ModemManager[672]: <info>  [device /sys/devices/pci0000:00/0000:00:14.0/usb3/3-2] creating modem with plugin 'fibocom' and '3' ports
maj 06 18:23:41 ModemManager[672]: <info>  [base-manager] modem for device '/sys/devices/pci0000:00/0000:00:14.0/usb3/3-2' successfully created
maj 06 18:23:41 ModemManager[672]: opening device...
maj 06 18:23:41 ModemManager[672]: [/dev/cdc-wdm0] Read max control message size from descriptors file: 512
maj 06 18:23:41 ModemManager[672]: <info>  [modem0/cdc-wdm0/mbim] MBIM device is not QMI capable
maj 06 18:23:41 ModemManager[672]: <warn>  [modem0] couldn't load carrier config: QMI PDC not supported
maj 06 18:23:41 ModemManager[672]: <warn>  [modem0] (shared-xmm) unexpected band value: 0-2
maj 06 18:23:41 ModemManager[672]: <warn>  [modem0] couldn't query SIM slots: OperationNotAllowed
maj 06 18:23:41 ModemManager[672]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1f.6': not supported by any plugin
maj 06 18:23:41 ModemManager[672]: <info>  [base-manager] couldn't check support for device '/sys/devices/pci0000:00/0000:00:1c.4/0000:09:00.0': not supported by any plugin
maj 06 18:23:44 ModemManager[672]: <warn>  [modem0/sim0] couldn't load list of emergency numbers: Failed to parse CRSM query result '+CRSM: 105,129'
maj 06 18:23:44 ModemManager[672]: <warn>  [modem0/sim0] couldn't load list of preferred networks: Not found
maj 06 18:23:45 ModemManager[672]: <warn>  [modem0] couldn't load initial EPS bearer settings: LTE attach status info is unsupported
maj 06 18:23:45 ModemManager[672]: <warn>  [modem0] couldn't load SUPL server: Operation not supported
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] state changed (unknown -> disabled)
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] state changed (disabled -> enabling)
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] power state updated: on
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] state changed (enabling -> enabled)
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] 3GPP registration state changed (unknown -> registering)
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] 3GPP registration state changed (registering -> home)
maj 06 18:23:45 ModemManager[672]: <info>  [modem0] state changed (enabled -> registered)
```

**Comment by yuwata** [comment]
Could you provide the results of `udevadm info` for relevant devices on both v250 and v251-rc2?

@aleksander0m Please let me know if you have any ideas.

**Comment by galibozek** [comment]
`251rc2`

```P: /devices/pci0000:00/0000:00:14.0/usb3/3-2/3-2:1.0/wwan/wwan0/wwan0mbim0
M: wwan0mbim0
R: 0
U: wwan
T: wwan_port
D: c 240:0
N: wwan0mbim0
L: 0
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb3/3-2/3-2:1.0/wwan/wwan0/wwan0mbim0
E: DEVNAME=/dev/wwan0mbim0
E: DEVTYPE=wwan_port
E: MAJOR=240
E: MINOR=0
E: SUBSYSTEM=wwan
E: USEC_INITIALIZED=9488568
E: ID_MM_CANDIDATE=1
E: ID_MM_PORT_TYPE_MBIM=1

P: /devices/pci0000:00/0000:00:14.0/usb3/3-2
M: 3-2
R: 2
U: usb
T: usb_device
D: c 189:257
N: bus/usb/003/002
L: 0
V: usb
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb3/3-2
E: SUBSYSTEM=usb
E: DEVNAME=/dev/bus/usb/003/002
E: DEVTYPE=usb_device
E: DRIVER=usb
E: PRODUCT=2cb7/210/333
E: TYPE=239/2/1
E: BUSNUM=003
E: DEVNUM=002
E: MAJOR=189
E: MINOR=257
E: USEC_INITIALIZED=8299987
E: ID_VENDOR=FIBOCOM
E: ID_VENDOR_ENC=FIBOCOM
E: ID_VENDOR_ID=2cb7
E: ID_MODEL=L830-EB-00
E: ID_MODEL_ENC=L830-EB-00
E: ID_MODEL_ID=0210
E: ID_REVISION=0333
E: ID_SERIAL=FIBOCOM_L830-EB-00_004999010640000
E: ID_SERIAL_SHORT=004999010640000
E: ID_BUS=usb
E: ID_USB_INTERFACES=:020e00:0a0002:020200:0a0000:
E: ID_VENDOR_FROM_DATABASE=Fibocom
E: ID_MODEL_FROM_DATABASE=L830-EB-00 LTE WWAN Modem
E: ID_PATH=pci-0000:00:14.0-usb-0:2
E: ID_PATH_TAG=pci-0000_00_14_0-usb-0_2

P: /devices/pci0000:00/0000:00:14.0/usb3/3-2/3-2:1.0/usbmisc/cdc-wdm0
M: cdc-wdm0
R: 0
U: usbmisc
D: c 180:176
N: cdc-wdm0
L: 0
E: DEVPATH=/devices/pci0000:00/0000:00:14.0/usb3/3-2/3-2:1.0/usbmisc/cdc-wdm0
E: DEVNAME=/dev/cdc-wdm0
E: MAJOR=180
E: MINOR=176
E: SUBSYSTEM=usbmisc
E: USEC_INITIALIZED=9436367
E: ID_MM_CANDIDATE=1
```

`250.5`

```soon
```

**Comment by yuwata** [comment]
Hm, it seems `SUBSYSTEMS=="usb"` does not work on v251...

**Comment by yuwata** [comment]
@galibozek Thank you for the report. Fix is waiting in #23291.

**Comment by galibozek** [comment]
Commit: eba782d59bfc1ce82b4c41883d15fa5e705126e5 resolved my problem, thanks
